### PR TITLE
Switch to MKDocs

### PR DESCRIPTION
**Pull request by waylan**
This is a work in progress.

I merged in 6f87b32 from the md3 branch. That branch has all of the
extension docs removed, so they haven't been converted yet. And I
only resolved conflicts and updated mkdocs.yml to the current MkDocs
format.

Addresses #601.

### CODE REVIEW

**Comment by waylan** [comment]
I still need to do an audit of the docs to make sure I didn't miss anything, but I pushed the deployment up to GitHub Pages. However, weirdly the homepage is not showing unless I include `index.html` in the URL. So this works https://python-markdown.github.io/index.html but this doesn't https://python-markdown.github.io/. Even more strange, this works fine https://python-markdown.github.io/extensions/. Not sure what's going on with the homepage there. Unless its some weird caching thing.

**Comment by facelessuser** [comment]
https://python-markdown.github.io/ works for me

**Comment by facelessuser** [comment]
Just my 2 cents, but links are hard to see in that theme.  If you can make them stand out a little, it might help with navigation.

**Comment by waylan** [comment]
The homepage works for me from a different location/machine. Must have been a caching thing causing the problem.

> Just my 2 cents, but links are hard to see in that theme. If you can make them stand out a little, it might help with navigation.

The CSS is completely unmodified from the Sphinx theme. That is the "problem" with navigation? I didn't have any problems, but then I'm intimately aware of the structure of the theme. 

I did notice that tables seem to be completely unstyled, so that may need to be investigated and possibly addressed. But that would likely happen upstream on the theme, not here.

Regardless, at this point I'm mostly concerned with the MkDocs config, whether all internal links are properly converted, and if the rendered Markdown works with the theme (for example, some code blocks might need to have a language identified as language guessing appears to be getting it wrong). I'll be doing an audit page-by-page, so if anyone sees anything, let me know in a comment and I'll be sure to address it.

**Comment by facelessuser** [comment]
> The CSS is completely unmodified from the Sphinx theme. That is the "problem" with navigation? I didn't have any problems, but then I'm intimately aware of the structure of the theme.

Dark navy blue and black is hard to distinguish...at least for me (at first).  I had to mouse around to get the link to turn red on hover to see where they were.  Once I stared at it a bit, I more easily noticed the subtle differences, but I'd argue they are a bit *too* subtle.  It won't really affect me much as I don't often have to reference the docs, just my general impression.  Lightening it up to `#007fb4` makes them easy to see, but even `#006d9b` would do it less though but give it a bit more contrast.

It's up to you, just figured I'd mention it, I'll deal regardless.  After a bit my eyes can see the difference.  I just personally dislike sites where clickable things are too subtle (this happens a lot with the flat theme trend).  I feel like I just flail my mouse around until I see the cursor or the link change to some easy to see hover state.

**Comment by waylan** [comment]
After a lot of cleanup, I merged this. We can deal with theme related issues separately.

File: makefile [hunk 1]
```diff
@@ -22,6 +22,7 @@ install:
 deploy:
 	python setup.py sdist --manifest-only
 	python setup.py sdist --formats zip,gztar upload
+	mkdocs gh-deploy -r pages -b master
```
Comment (waylan) [inline makefile:25]: Turns out this doesn't work. It hoses your local master branch the first time and then (after more commits are added) results in merge conflicts. See davisp/ghp-import#65 for details.

**Pull request merged**